### ISSUE TITLE
Allow to provide custom column name for transformed feature

### ISSUE THREAD

**Issue by ragrawal** [pandas-dev/sklearn-pandas#83]
I have a datetime values in a column, "submitted_at", and I am trying to construct two features out of it, namely: day of week and hour of day. However the output dataframe contains "submitted_at" twice. Once for day of week feature and another for hour of day feature.

**Comment by dukebody** [comment]
Hi @ragrawal . Can you post the code you are using? You can always change afterwards the name of the output columns to the ones you want.

**Comment by dukebody** [comment]
Thanks for the contribution! I merged a branch that includes your commit with some tests+docs. See https://github.com/paulgb/sklearn-pandas/commit/2fc6286e2d602a1e1b7d0eab7c256f006eca58ac

**Comment by dukebody** [comment]
Sorry, didn't really read the title of the issue. What do you think about an API like:

```
mapper = DataFrameMapper([
    ('target', Transformer1(), 'transformed_target_1'),
    ('target', Transformer2(), 'transformed_target_2')
])
```

Meaning that the result of applying `Transformer1` to the column `target` should be placed in a column with name `transformed_target_1`, and equivalent for the second one?

This is probably quite easy to do by passing the chosen names to the `get_names` method, if present.

**Comment by ragrawal** [comment]
submitted a pull request related to this: https://github.com/paulgb/sklearn-pandas/pull/91

**Comment by dukebody** [comment]
Fixed in #92.

**Comment by macwanjason** [comment]
Is there a way to rename multiple column outputs at a time? For example:

```
import pandas as pd
from sklearn.base import TransformerMixin
from sklearn_pandas import DataFrameMapper



data2 = pd.DataFrame({'dates1': ['2001-12-20','2002-10-21','2003-08-22','2004-08-23', 
                                 '2004-07-20','2007-12-21','2006-12-22','2003-04-23'],   
                     'dates2'  : ['2012-12-20','2009-10-21','2016-08-22','2017-08-23', 
                                 '2014-07-20','2011-12-21','2014-12-22','2015-04-23']})

class DateFormatter(TransformerMixin):

    def fit(self, X, y=None):
        # stateless transformer
        return self

    def transform(self, X):
        # assumes X is a DataFrame
        Xdate = X.apply(pd.to_datetime)
        return Xdate


class DateEncoder(TransformerMixin):

    def fit(self, X, y=None):
        return self

    def transform(self, X):
        dt = X.dt
        return pd.concat([dt.year, dt.month, dt.day], axis=1)


DATE_COLS = ['dates1', 'dates2']

datemult = DataFrameMapper(
            [ (i,[DateFormatter(),DateEncoder()]) for i in DATE_COLS     ] 
            , input_df=True, df_out=True)

df = datemult.fit_transform(data2)
```
This outputs:
```
Out[4]: 
   dates1_0  dates1_1  dates1_2  dates2_0  dates2_1  dates2_2
0      2001        12        20      2012        12        20
1      2002        10        21      2009        10        21
2      2003         8        22      2016         8        22
3      2004         8        23      2017         8        23
4      2004         7        20      2014         7        20
5      2007        12        21      2011        12        21
6      2006        12        22      2014        12        22
7      2003         4        23      2015         4        23
```

But I would very much like to rename , say, dates_1_0 to dates_1_year or dates_2_1 to dates_2_month
